docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/FastTree.cs

```diff
@@ -20,18 +20,48 @@ public static void Example()
             var examples = GenerateRandomDataPoints(1000);
 
             // Convert the examples list to an IDataView object, which is consumable by ML.NET API.
-            var data = mlContext.Data.LoadFromEnumerable(examples);
+            var trainingData = mlContext.Data.LoadFromEnumerable(examples);
 
             // Define the trainer.
-            var pipeline = mlContext.BinaryClassification.Trainers.FastTree();
+            var pipeline = mlContext.Regression.Trainers.FastTree();
 
             // Train the model.
-            var model = pipeline.Fit(data);
+            var model = pipeline.Fit(trainingData);
+
+            // Create testing examples. Use different random seed to make it different from training data.
+            var testData = mlContext.Data.LoadFromEnumerable(GenerateRandomDataPoints(500, seed:123));
+
+            // Run the model on test data set.
+            var transformedTestData = model.Transform(testData);
+
+            // Convert IDataView object to a list.
+            var predictions = mlContext.Data.CreateEnumerable<Prediction>(transformedTestData, reuseRowObject: false).ToList();
+
+            // Look at 5 predictions
+            foreach (var p in predictions.Take(5))
+                Console.WriteLine($"Label: {p.Label:F3}, Prediction: {p.Score:F3}");
+
+            // Expected output:
+            //   Label: 0.985, Prediction: 0.938
+            //   Label: 0.155, Prediction: 0.131
+            //   Label: 0.515, Prediction: 0.517
+            //   Label: 0.566, Prediction: 0.519
+            //   Label: 0.096, Prediction: 0.089
+
+            // Evaluate the overall metrics
+            var metrics = mlContext.Regression.Evaluate(transformedTestData);
+            SamplesUtils.ConsoleUtils.PrintMetrics(metrics);
+
+            // Expected output:
+            //   Mean Absolute Error: 0.05
+            //   Mean Squared Error: 0.00
+            //   Root Mean Squared Error: 0.06
+            //   RSquared: 0.95
         }
 
-        private static IEnumerable<DataPoint> GenerateRandomDataPoints(int count)
+        private static IEnumerable<DataPoint> GenerateRandomDataPoints(int count, int seed=0)
```

Comment:
> private static IEnumerable<DataPoint> GenerateRandomDataPoints(int count, int seed=0) [](start = 7, length = 86)

i like this, but do you think it will be quick  to create them artificially for each task? Ranking and time series come to mind.

Reply:
sometimes it's not easy, ranking being an example. For those, I'll keep the text-loader style.

so this template is mostly suitable for regression and binary classification.

---
In reply to: [266151216](https://github.com/dotnet/machinelearning/pull/2979#discussion_r266151216) [](ancestors = 266151216)

Reply:
Text data is also not easy to randomly generated.

---
In reply to: [266157156](https://github.com/dotnet/machinelearning/pull/2979#discussion_r266157156) [](ancestors = 266157156,266151216)

Reply:
I believe Zeeshan A meant meaningful text data but I don't think we need meaningful data to demonstrate the functionality of a module. The amount of data might be a problem to trainers, but to my knowledge, there is no trainer directly consuming strings.

Reply:
This sample is just a template. If in some scenarios it doesn't make sense, we can try text-loader instead.

---
In reply to: [266160094](https://github.com/dotnet/machinelearning/pull/2979#discussion_r266160094) [](ancestors = 266160094,266157156,266151216)